### ISSUE TITLE
Documentation display is broken

### ISSUE THREAD

**Issue by LaurentTreguier** [dlang-vscode/dlang-vscode#28]
Currently, documentation is not always displayed properly.
Some characters used in markdown are not escaped. For example, any `*` pair will be interpreted by VSCode and the text in between the two `*` will be shown in italic.
Some parts of the documentation use a `$(UL something)` syntax that are only translated to ``` `something` ```, when it should be something else entirely (here with `UL` it should be a list with some `LI` elements inside).
Furthermore, something like `$(UL text (parens) other text)` will be replaced by the simple regex with ``` `text (parens` other text)``` because it stops at the first closing parenthesis.

Long story short, the parsing done for documentation should be a lot more complex than it currently is (a couple of regex replacements and separation of text and code examples).
